oauth2-useragent-core/src/main/java/com/microsoft/alm/oauth2/useragent/AuthorizationResponse.java

```diff
@@ -50,19 +50,27 @@ public String getState() {
     }
 
     public static AuthorizationResponse fromString(final String s) throws AuthorizationException {
+        return fromString(s, null);
+    }
+
+    public static AuthorizationResponse fromString(final String s, final String potentialDescription) throws AuthorizationException {
         String code = null;
         String state = null;
         String error = "unknown_error";
-        String errorDescription = null;
+        String errorDescription = potentialDescription;
         String errorUriString = null;
-        if (s != null) {
+        if (s != null && s.length() > 0) {
             final String[] pairs = PAIR_SEPARATOR.split(s.trim());
 
             for (final String pair : pairs) {
                 final String[] nameAndValue = NAME_VALUE_SEPARATOR.split(pair, 2);
                 try {
                     if (nameAndValue.length != 2) {
-                        throw new AuthorizationException("parsing_error", "Failed to parse server response", null, null);
+                        final StringBuilder sb = new StringBuilder("Failed to parse server response");
+                        if (errorDescription != null) {
+                            sb.append(".\nDetails: ").append(errorDescription);

```

Comment:
> Agree it's helpful, but I am afraid we may include too much -- if it contains sensitive information (like the code, or part of it)?

I don't think there's anything sensitive there; even if a user copy-pastes a [garbled] string that included the authorization code, those are only supposed to be valid for a very short time and for a very specific request.  I expect it will be very rare that we encounter a string that we can partially parse.